source/paddle.js

```diff
@@ -889,6 +1006,503 @@ paddle.AttributeType = {
     FLOAT64: 15
 };
 
+paddle.IR = class {
+
+    constructor(obj) {
+        this.base_code = obj.base_code;
+        this.version = obj.base_code.version;
+
+        const globalInfo = new paddle.IR.GlobalInfo();
+        this.program = new paddle.IR.Program(obj.program, globalInfo);
```

Comment:
Collapse `paddle.IR`, `paddle.IR.GlobalInfo` and `paddle.IR.Program` into a single `paddle.IrReader` class.

Reply:
Done!